lib/hal/video.c

```diff
@@ -230,6 +236,24 @@ BOOLEAN XVideoListModes(VIDEO_MODE *vm, int bpp, int refresh, void **p)
 		if(pVidMode->refresh != refresh)
 			continue;
 
+		if(dwAdapter == AV_PACK_HDTV)
+		{
+			is_hd = pVidMode->dwMode & 0x80000000;
+
+			if(pVidMode->height == 720 && !allow_720p)
+				continue;
+
+			if(pVidMode->height == 1080 && !allow_1080i)
+				continue;
```

Comment:
720p and 1080i mode should follow 480 modes so you get:
- 480i rejection + including comment
- 480p check
- 720p check
- 1080i check

Reply:
Fixed. :+1: